webapi/Lib/WebHooks/MccSoft.WebHooks/Manager/WebHookManager.cs

```diff
@@ -0,0 +1,142 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Hangfire;
+using Hangfire.Common;
+using MccSoft.WebHooks.Domain;
+using MccSoft.WebHooks.Processing;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace MccSoft.WebHooks.Manager;
+
+public class WebHookManager<TSub> : IWebHookManager<TSub>
+    where TSub : WebHookSubscription
+{
+    private readonly ILogger<WebHookManager<TSub>> _logger;
+    private readonly DbContext _dbContext;
+    private readonly IServiceScopeFactory _serviceScopeFactory;
+
+    public WebHookManager(
+        IServiceScopeFactory serviceScopeFactory,
+        IServiceProvider serviceProvider,
+        ILogger<WebHookManager<TSub>> logger
+    )
+    {
+        if (WebHookRegistration.DbContextType == null)
+            throw new ArgumentNullException(
+                $"WebHookRegistration.DbContextType wasn't initialized. Did you call builder.AddWebHookEntities() from DbContext OnModelCreating?"
+            );
+
+        _serviceScopeFactory = serviceScopeFactory;
+        _dbContext = (DbContext)
+            serviceProvider.GetRequiredService(WebHookRegistration.DbContextType);
+
+        _logger = logger;
+    }
+
+    /// <inheritdoc />
+    public async Task<TSub> GetSubscriptionAsync(Guid subscriptionId)
+    {
+        return await _dbContext
+            .WebHookSubscriptions<TSub>()
+            .FirstAsync(x => x.Id == subscriptionId);
+    }
+
+    /// <inheritdoc />
+    public async Task<IEnumerable<TSub>> GetSubscriptionsAsync(bool withAttempts = false)
```

Comment:
maybe return IQueryable here?

Reply:
I'll double check. I'm afrad that we could call `.ToList()`  somewhere